core/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java

```diff
@@ -71,4 +141,64 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String getName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        MultiTermQuery.RewriteMethod method = QueryParsers.parseRewriteMethod(rewrite, null);
+
+        Query query = null;
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            query = fieldType.prefixQuery(value, method, parseContext);
+        }
+        if (query == null) {
+            PrefixQuery prefixQuery = new PrefixQuery(new Term(fieldName, BytesRefs.toBytesRef(value)));
+            if (method != null) {
+                prefixQuery.setRewriteMethod(method);
+            }
+            query = prefixQuery;
+        }
+        
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (Strings.isEmpty(this.fieldName)) {
+            validationException = addValidationError("field name cannot be null or empty.", validationException);
+        }
+        if (this.value == null) {
+            validationException = addValidationError("query text cannot be null", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    protected PrefixQueryBuilder createBuilder(String fieldName, Object value) {
+        return new PrefixQueryBuilder(fieldName, value);
+    }
+
+    @Override
+    protected PrefixQueryBuilder doReadFrom(StreamInput in) throws IOException {
+        PrefixQueryBuilder prefixQueryBuilder = super.doReadFrom(in);
+        prefixQueryBuilder.rewrite = in.readOptionalString();
+        return prefixQueryBuilder;
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        super.doWriteTo(out);
+        out.writeOptionalString(rewrite);
+    }
+
+    @Override
+    protected final int doHashCode() {
+        return super.doHashCode() + Objects.hash(rewrite);
+    }
+
+    @Override
+    protected boolean doEquals(PrefixQueryBuilder other) {
+        return super.doEquals(other) && Objects.equals(rewrite, other.rewrite);
+    }
```

Comment:
I am slightly conflicted here on how much code we saved here by overriding BaseTermQueryBuilder... just declaration of fieldName and value plus getters and setters and potentially validate?